test/test_prototype_transforms_functional.py

```diff
@@ -4,188 +4,233 @@
-
+class TestKernels:
+    sample_inputs = pytest.mark.parametrize(
+        ("info", "args_kwargs"),
+        [
+            pytest.param(info, args_kwargs, id=f"{info.kernel_name}-{idx}")
+            for info in KERNEL_INFOS
+            for idx, args_kwargs in enumerate(info.sample_inputs_fn())
+        ],
+    )
 
-def _get_distinct_kernels():
-    kernel_names = {
-        name
-        for name, f in F.__dict__.items()
-        if callable(f) and not name.startswith("_") and any(name.endswith(k) for k in _KERNEL_TYPES)
-    }
-    return _distinct_callables(kernel_names)
+    @sample_inputs
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_scripted_vs_eager(self, info, args_kwargs, device):
+        kernel_eager = info.kernel
+        kernel_scripted = script(kernel_eager)
 
+        args, kwargs = args_kwargs.load(device)
 
-def _get_distinct_midlevels():
-    midlevel_names = {
-        name
-        for name, f in F.__dict__.items()
-        if callable(f) and not name.startswith("_") and not any(name.endswith(k) for k in _KERNEL_TYPES)
-    }
-    return _distinct_callables(midlevel_names)
+        actual = kernel_scripted(*args, **kwargs)
+        expected = kernel_eager(*args, **kwargs)
 
+        assert_close(actual, expected, **info.closeness_kwargs)
 
-@pytest.mark.parametrize(
-    "kernel",
-    [
-        pytest.param(kernel, id=name)
-        for name, kernel in _get_distinct_kernels()
-        if not name.endswith("_image_pil") and name not in {"to_image_tensor"}
-    ],
-)
-def test_scriptable_kernel(kernel):
-    jit.script(kernel)  # TODO: pass data through it
+    # TODO: We need this until the kernels below also have `KernelInfo`'s. If they do, `test_scripted_vs_eager` replaces
+    #  this test for them.
+    @pytest.mark.parametrize(
+        "kernel",
+        [
+            F.adjust_brightness_image_tensor,
+            F.adjust_gamma_image_tensor,
+            F.adjust_hue_image_tensor,
+            F.adjust_saturation_image_tensor,
+            F.clamp_bounding_box,
+            F.five_crop_image_tensor,
+            F.normalize_image_tensor,
+            F.ten_crop_image_tensor,
+        ],
+        ids=lambda kernel: kernel.__name__,
+    )
+    def test_scriptable(self, kernel):
+        script(kernel)
+
+    @sample_inputs
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_batched_vs_single(self, info, args_kwargs, device):
+        def unbind_batch_dims(batched_tensor, *, data_dims):
+            if batched_tensor.ndim == data_dims:
+                return batched_tensor
+
+            return [unbind_batch_dims(t, data_dims=data_dims) for t in batched_tensor.unbind(0)]
+
+        def stack_batch_dims(unbound_tensor):
+            if isinstance(unbound_tensor[0], torch.Tensor):
+                return torch.stack(unbound_tensor)
+
+            return torch.stack([stack_batch_dims(t) for t in unbound_tensor])
```

Comment:
Shouldn't we define once this functions instead of on each execution ?

Reply:
Can do.